pygmt/src/plot.py

```diff
@@ -209,6 +209,8 @@ def plot(self, x=None, y=None, data=None, size=None, direction=None, **kwargs):
     extra_arrays = []
     if "S" in kwargs and kwargs["S"][0] in "vV" and direction is not None:
         extra_arrays.extend(direction)
+    elif "S" not in kwargs and kind == "geojson" and (data.geom_type == "Point").all():
```

Comment:
> 1. the current implementation works only with Point, I don't think we should support mixed geometrys in geopandas DataFrame it's a bad practice. we can add support for MultiPoint alon with `(data.geom_type == "Point").all() or (data.geom_type == "MultiPoint").all()`

Agree with not supporting mixed geometries. Let's go with using `data.geom_type.isin(["Point", "MultiPoint"]).all()` (see [`pandas.Series.isin` ](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.isin.html)) which will do the same as the `or` operator.

```suggestion
    elif "S" not in kwargs and kind == "geojson" and data.geom_type.isin(["Point", "MultiPoint"]).all():
```

> 2. We don't need to worry about the execution time of `(data.geom_type == "Point").all()` because the execution time of writing the temporary file is larger (probably also plotting the figure but I didn't check)

Good to have those numbers! 1 millisecond is acceptable. I've tried to check both Point and MultiPoint and the timings aren't too bad either.

```python
%%timeit
gdf.geom_type.isin(["Point", "MultiPoint"]).all()
# 1.33 ms ± 13.2 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```